ssc/cmod_pvwattsv7.cpp

```diff
@@ -0,0 +1,1261 @@
+			double R_sh = m.R_shref + (m.R_sh0 - m.R_shref) * exp(-m.R_shexp * (S / S_ref));
+			double V_oc = openvoltage_5par_rec(m.Voc, a, I_L, I_0, R_sh, m.D2MuTau, Vbi);
+
+			double V, I;
+			return maxpower_5par_rec(V_oc, a, I_L, I_0, m.R_s, R_sh, m.D2MuTau, Vbi, &V, &I);
+		}
+		else
+			return 0.0;
+	}
+
+	void exec() throw(general_error)
+	{
+		// don't add "gen" output if battery enabled, gets added later
+		if (!as_boolean("batt_simple_enable"))
+			add_var_info(vtab_technology_outputs);
+
+		std::unique_ptr<weather_data_provider> wdprov;
+
+		if (is_assigned("solar_resource_file"))
+		{
+			const char *file = as_string("solar_resource_file");
+			wdprov = std::unique_ptr<weather_data_provider>(new weatherfile(file));
+
+			weatherfile *wfile = dynamic_cast<weatherfile*>(wdprov.get());
+			if (!wfile->ok()) throw exec_error("pvwattsv7", wfile->message());
+			if (wfile->has_message()) log(wfile->message(), SSC_WARNING);
+		}
+		else if (is_assigned("solar_resource_data"))
+		{
+			wdprov = std::unique_ptr<weather_data_provider>(new weatherdata(lookup("solar_resource_data")));
+		}
+		else
+			throw exec_error("pvwattsv7", "no weather data supplied");
+
+		pv.dc_nameplate = as_double("system_capacity") * 1000;
+		pv.dc_ac_ratio = as_double("dc_ac_ratio");
+		pv.ac_nameplate = pv.dc_nameplate / pv.dc_ac_ratio;
+
+		pv.xfmr_rating = pv.ac_nameplate;
+
+		pv.inv_eff_percent = as_double("inv_eff");
+
+		size_t soiling_len = 0;
+		ssc_number_t *soiling = nullptr;
+		if (is_assigned("soiling"))
+		{
+			soiling = as_array("soiling", &soiling_len);
+		}
+
+		size_t albedo_len = 0;
+		ssc_number_t *albedo = 0;
+		if (is_assigned("albedo"))
+		{
+			albedo = as_array("albedo", &albedo_len);
+		}
+
+		pv.dc_loss_percent = as_double("losses");
+		pv.tilt = pv.azimuth = std::numeric_limits<double>::quiet_NaN();
+		pv.rotlim = 45.0;
+		if (is_assigned("tilt")) pv.tilt = as_double("tilt");
+		if (is_assigned("azimuth")) pv.azimuth = as_double("azimuth");
+		if (is_assigned("rotlim")) pv.rotlim = as_double("rotlim");
+
+		pv.xfmr_ll_f = as_double("xfmr_ll")*0.01; //transformer inputs always present, but default to 0
+		pv.xfmr_nll_f = as_double("xfmr_nll")*0.01;
+
+		bool enable_wind_stow = as_boolean("enable_wind_stow");
+		double wstow = std::numeric_limits<double>::quiet_NaN();
+		if (is_assigned("stow_wspd")) wstow = as_double("stow_wspd"); // wind stow speed, m/s.
+		double gustf = std::numeric_limits<double>::quiet_NaN(); // gust factor
+		if (is_assigned("gust_factor")) gustf = as_double("gust_factor");
+		double wind_stow_angle_deg = 30; // assume stowing at 30 degrees for better dynamic torsional stability, despite higher static loading on piles
+		if (is_assigned("wind_stow_angle")) wind_stow_angle_deg = as_double("wind_stow_angle");
+
+		int en_sdm = is_assigned("en_sdm") ? as_integer("en_sdm") : 0; //whether or not to use the mermoud lejeune single diode model as defined above
```

Comment:
It's essentially a hidden input variable- it can still be set but shouldn't appear as an option to most people, which I think is appropriate for this particular input. However, explained that better with a comment in 4ea820c9eca375d9c009fbef945246f20884c8d9 :)